src/services/plasma.js

```diff
@@ -0,0 +1,118 @@
+import web3 from "web3";
+import { getStoredValue, storeValues } from "./localStorage";
+
+const COUNTRIES = {
+  US_OF_AMBROSIA: {
+    color: 49156,
+    address: "0x3378420181474D3aad9579907995011D6a545E3D",
+    fullName: "United States of Ambrosia",
+    shortName: "USA"
+  },
+  US_OF_BALOONS: {
+    color: 49155,
+    address: "0x181fc600915c35F4e44d41f9203A7c389b4A7189",
+    fullName: "United States of Baloons",
+    shortName: "USB"
+  }
+};
+
+export const sliceHex = (hexString, start = 0, end = hexString.length) => {
+  return `0x${hexString
+    .slice(2) // strip 0x
+    .slice(start * 2, end * 2)}`;
+};
+export const extractData = passport => {
+  const rawData = passport.output.data;
+
+  const nameHex = sliceHex(rawData, 0, 20);
+  const imageHex = sliceHex(rawData, 20, 24);
+  const emittedHex = sliceHex(rawData, 24, 28);
+  const lockedHex = sliceHex(rawData, 28, 32);
+
+  const { hexToString, hexToNumber } = web3.utils;
+
+  const name = hexToString(nameHex);
+  const image = hexToString(imageHex);
+  const emitted = hexToNumber(emittedHex);
+  const locked = hexToNumber(lockedHex);
+
+  return {
+    name,
+    image,
+    emitted,
+    locked
+  };
+};
+export const getId = passport => passport.output.value;
+
+export const getCitizenship = async (plasma, account) => {
+  const citizenship = getStoredValue("citizenship", account);
+  if (!citizenship) {
+    fetchPassportData(plasma, account);
+  } else {
+    return citizenship;
+  }
+};
+
+export const fetchAlPassports = async (plasma, account) => {
```

Comment:
typo: missing an `l` → `fetchAllPassports`

Reply:
Nice spot! Fixed that one ;)